colossalai/inference/modeling/models/nopadding_llama.py

```diff
@@ -298,6 +298,10 @@ def __init__(
         self.gate_up_weight = nn.Parameter(
             torch.stack([mlp_gproj_w.transpose(0, 1), mlp_uproj_w.transpose(0, 1)], dim=0)
         )
+        self.gate_up_dict = {
+            "gate_proj.weight": None,
+            "up_proj.weight": None,
+        }  # used and delattr in load/shard of gate/up weight
```

Comment:
Wonder why we initialize the dictionary here? It seems you only use it when loading from state dict

Reply:
q,k,v will be stored in diff checkpoint files, we need to maintain a global dict to store the q,k,v. So hardcode is inevitable here :(.